File: cpp/392_isSubsequence.cpp
bool isSubsequence(string s, string t) {
        int n = s.size(),j=0;

        if(n>t.size())
            return false;


        for(int i=0;i<t.size();i++){
            if(j==n)
                break;
            if(t[i]==s[j])
                ++j;
        }

        if(j==n)
            return true;

        return false;
    }

File: py/392_isSubsequence.py
def isSubsequence(s:str, t:str)->bool:
    if len(s)==0:
        return True
    
    if len(t)==0:
        return False
    
    i:int = 0
    for j in range(len(t)):
        if t[j]==s[i]:
            i+=1

    if i<len(s) and i==len(s):
        return True
    
    return False